src/plugins/pinDms/index.tsx

```diff
@@ -1,133 +1,368 @@
+
+        return {
+            sections: this.sections,
+            chunkSize: this.getChunkSize(),
+        };
+    },
+
+    makeSpanProps() {
+        return {
+            onClick: () => this.collapseDMList(),
+            role: "button",
+            style: { cursor: "pointer" }
+        };
+    },
+
+    getChunkSize() {
+        // the chunk size is the amount of rows (measured in pixels) that are rendered at once (probably)
+        // the higher the chunk size, the more rows are rendered at once
+        // also if the chunk size is 0 it will render everything at once
+
+        const sections = this.getSections();
+        const sectionHeaderSizePx = sections.length * 40;
+        // (header heights + DM heights + DEFAULT_CHUNK_SIZE) * 1.5
+        // we multiply everything by 1.5 so it only gets unmounted after the entire list is off screen
+        return (sectionHeaderSizePx + sections.reduce((acc, v) => acc += v + 44, 0) + DEFAULT_CHUNK_SIZE) * 1.5;
+    },
+
+    getPinCount(channelIds: string[]) {
+        return channelIds.length ? this.getSections() : [];
+    },
+
+    isCategoryIndex(sectionIndex: number) {
+        return this.sections && sectionIndex > 0 && sectionIndex < this.sections.length - 1;
+    },
+
+    isChannelIndex(sectionIndex: number, channelIndex: number) {
+        if (settings.store.dmSectioncollapsed && sectionIndex !== 0)
+            return true;
+        const cat = categories[sectionIndex - 1];
+        return this.isCategoryIndex(sectionIndex) && (cat.channels.length === 0 || cat?.channels[channelIndex]);
+    },
+
+    isDMSectioncollapsed() {
+        return settings.store.dmSectioncollapsed;
+    },
+
+    collapseDMList() {
+        // console.log("HI");
+        settings.store.dmSectioncollapsed = !settings.store.dmSectioncollapsed;
+        forceUpdate();
+    },
+
+    isChannelHidden(categoryIndex: number, channelIndex: number) {
+        if (categoryIndex === 0) return false;
+
+        if (settings.store.dmSectioncollapsed && this.getSections().length + 1 === categoryIndex)
+            return true;
+
+        if (!this.instance || !this.isChannelIndex(categoryIndex, channelIndex)) return false;
+
+        const category = categories[categoryIndex - 1];
+        if (!category) return false;
+
+        return category.collapsed && this.instance.props.selectedChannelId !== category.channels[channelIndex];
+    },
+
+    getScrollOffset(channelId: string, rowHeight: number, padding: number, preRenderedChildren: number, originalOffset: number) {
+        if (!isPinned(channelId))
+            return (
+                (rowHeight + padding) * 2 // header
+                + rowHeight * this.getAllUncollapsedChannels().length // pins
+                + originalOffset // original pin offset minus pins
+            );
+
+        return rowHeight * (this.getAllUncollapsedChannels().indexOf(channelId) + preRenderedChildren) + padding;
+    },
+
+    renderCategory({ section }: { section: number; }) {
```

Comment:
use an ErrorBoundary
```suggestion
    renderCategory: ErrorBoundary.wrap(({ section }: { section: number; }) => {
```